addon/utils/domain-tweaks.js

```diff
@@ -0,0 +1,46 @@
+import { zip, union } from 'lodash';
+
+/**
+ * function to add custom ticks to the chart.
+ * @param {*} chart is the chart being referred - column-chart, barchart.
+ * @param {*} domain the un-padded domain.
+ * @param {*} otherTicks also add other ticks along with domain ticks because existing ticks are overwritten.
+ */
+export function addDomainTicks(chart, domain, otherTicks = undefined) {
+    if (!chart || !domain) {
+        return;
+    }
+
+    const yAxis = chart.yAxis && chart.yAxis();
+    if (!yAxis) {
+        return;
+    }
+
+    const yAxisTicks = yAxis.ticks && yAxis.ticks();
+
+    if (!yAxisTicks) {
+        return;
+    }
+
+    otherTicks = otherTicks || [];
+    yAxisTicks.tickValues(union(domain, otherTicks));
+}
+
+/**
+ * Adds extra padding to the domain to accomodate datavalues and min/max indicator on top of the bars/columns.
+ * @param {*} domain is a alist : [upperbound, lowerbound]
+ * @returns {*} an object {actualDomain: [lowerbound, upper], paddedDomain: [paddedlowerbound, paddedupperbound]}
+ */
+export function padDomain(domain) {
+    return zip(domain, ['lower', 'upper']).map(([d, indicator]) => {
```

Comment:
Since you know that `domain` should always be an array of length 2, you could remove `zip` here. Also, if the purpose of this utility function is to add +- 40%, could we just use the absolute value?

```
const lowerBound = domain[0];
const lowerPadding = Math.abs(lowerBound) * 0.4;

// I think you always want to bring the y value down
const paddedLower = lowerBound - lowerPadding;

// do the same for domain upperbound

return [paddedLower, paddedUpper];
```

Reply:
Also, it looks like `padDomain` returns an array `[paddedLower, paddedUpper]` and not an object `{ actualDomain: [lower, upper], paddedDomain: [lower, upper]}` as mentioned in the jsdoc

Reply:
simplified the function, and fixed jsdoc comment.